src/Logger.jl

```diff
@@ -111,3 +111,50 @@ function reset_tb_logs()
     logdir = default_logging_session[].logdir
     default_logging_session[] = Logger(logdir, overwrite=true)
 end
+
+
+# Implement the AbstractLogger Interface
+
+catch_exceptions(lg::TBLogger) = false
+
+min_enabled_level(lg::TBLogger) = lg.min_level
+
+# For now, log everything that is above the lg.min_level
+shouldlog(lg::TBLogger, level, _module, group, id) = true
+
+function handle_message(lg::TBLogger, level, message, _module, group, id, file, line; kwargs...)
+    # Unpack the message
+    summ    = SummaryCollection()
+    i_step = 1
+
+    if !isempty(kwargs)
+        for (key,val) in pairs(kwargs)
+            # special values
+            if key == :delta_step
+                i_step = val
+                continue
+            end
+
+            data = Stack{Pair{String,Any}}()
+            name = message*"/$key"
+            push!(data, name => val)
+            while !isempty(data)
+                name, val = pop!(data)
+                loggable(val) ? push!(summ.value, summary_impl(name, val)) : preprocess(name, val, data)
+            end
+        end
+    end
+    iter = increment_step(lg, i_step)
+    write_event(lg.file, make_event(lg, summ, step=iter))
+end
+
+loggable(::Any) = false
+
+function preprocess(name, val, data)
```

Comment:
I should check If I can dispatch only on types which are structs. Otherwise I could check if `isstructtype` and if false throw an error.